File: Problems/0007_Reverse_Integer/0007_Reverse_Integer.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {
    //understand 32-bit signed integer, reverse digits of an integer
    //wait no need not really
    var result = 0;
    var i = 1;
    var mySign = Math.sign(x);
    x = x * mySign;
    while(x>0){
        result = (result*10)+(x%10)*i;
        x = parseInt(x/10);
    }
    return result*mySign;

};

File: Problems/0007_Reverse_Integer/0007_Reverse_Integer.java
class Solution {
    public int reverse(int x) {
        StringBuilder stringX = new StringBuilder();
        int result;
        if(x<0){
            stringX.append(String.valueOf(Math.abs(x)));
            stringX.reverse();
            try {
                result = Integer.parseInt(stringX.toString())*-1;
            }
            catch (NumberFormatException e) {
                return 0;
            }
            return result;
        }
        stringX.append(String.valueOf(x));
        stringX.reverse();
        try {
            result = Integer.parseInt(stringX.toString());
        }
        catch (NumberFormatException e) {
            return 0;
        }
        return result;
        
    }
}